lib/db/Repository.js

```diff
@@ -97,18 +97,30 @@ class Repository {
     });
   }
 
+  static aggregate(modelName, operatorArray) {
+    return Promise.try(() => {
+      const Model = Mongoose.model(modelName);
+      return Model
+        .aggregate(operatorArray);
+    });
+  }
+
   static search(modelName, searchCriteria, paginateOpts) {
     logger.debug(`Search on : ${modelName} with searchCriteria : ${JSON.stringify(searchCriteria)}`);
     const Model = Mongoose.model(modelName);
     let populateOptions;
     let sortBy;
     let searchBy;
+    let selectBy;
 
-    if (searchCriteria &&
-      (searchCriteria.populateOptions || searchCriteria.sortBy || searchCriteria.searchBy)) {
+    if (
+      searchCriteria &&
+      (searchCriteria.populateOptions || searchCriteria.sortBy || searchCriteria.searchBy || searchCriteria.selectBy)
+    ) {
       populateOptions = searchCriteria.populateOptions;
       sortBy = searchCriteria.sortBy;
       searchBy = searchCriteria.searchBy;
+      selectBy = searchCriteria.selectBy;
```

Comment:
can you rename this to `projection` or `responseAttrsToProject` or `responseAttrsToReturn` - 
`searchBy` - usage in context of `searchCriteria` misleads as this is not truly used in searchCritiera but you are using this field to project the response attribs.